### ISSUE TITLE
Window title bar counting towards canvas size

### ISSUE THREAD

**Issue by dabreegster** [dabreegster/abstreet#65]
Someone reports: "On Windows 10 2004, 3840x2160 monitor with 175% zoom, with the game maximized, the hitboxes for all the buttons are vertically offset lower than they should be by the height of the title bar of the window"

I need to check how winit reports window size; I remember seeing an outer/inner distinction

**Comment by dabreegster** [comment]
Also on a friend's Mac, when the game is fullscreen, moving the mouse to the top doesn't make the title bar drop down, which is normal for Mac. May be related.

**Comment by dabreegster** [comment]
Maybe related to https://github.com/rust-windowing/winit/issues/1516. Trying out a fix to ignore the size from resize events and trust inner size instead.

**Comment by jaredkrinke** [comment]
I commented on news.ycombinator.com, but it's probably better to move the discussion here, so copy-pasting:

That new build did NOT fix the issue for me.

Now that I'm looking more closely, this looks like some sort of initial aspect ratio issue, possibly related to a resize.

The game launches and the window either starts maximized or it maximizes itself. If I restore (un-maximize) and re-maximize, the aspect ratio is clearly different because the logo in the center of the screen gets slightly taller (while staying the same width).

In other words, the initial aspect ratio of the graphics seems incorrect (squished vertically slightly for me, running at 1920x1080).

**Comment by dabreegster** [comment]
Someone from HN had a thought: "Do you have your windows taskbar in non-default position? Mine is on top of screen and some games in some display modes give offset mouse position"

I think I misunderstood the problem yesterday; maybe all of the mouse events are just wrong. I dug out an old Windows laptop so I can test directly. I'll see what `WindowEvent::CursorMoved` actually has

**Comment by dabreegster** [comment]
Video from the other bug: https://imgur.com/a/ZcJKEx6

**Comment by jaredkrinke** [comment]
That other bug video looks identical to what I see. Also (see my later note on aspect ratio) note that the buttons in that video are 36x34 (i.e. not quite square / squished vertically).

My taskbar is in the default position (bottom of the screen), so I don't think that's related (also since the problem goes away if I un-maximize and re-maximize the problem goes away, it's only the initial condition that doesn't work correctly).

Also, just to draw attention to my last comment, I eventually noticed that this doesn't look like just a "mouse input" problem, the aspect ratio for rendering is off when I first launch. I'm not familiar with Rust or the window management library (or graphics library) you're using, but since I saw you're using a canvas, I wonder if the dimensions of your canvas are actually set incorrectly on launch for whatever reason (and re-maximizes fixes the problem).

Hope that helps!

**Comment by dabreegster** [comment]
Could somebody please try https://github.com/dabreegster/abstreet/suites/832299173/artifacts/9306832? Assuming the issue is still there, send the full output.txt -- it has some debug statements to figure out the window size upon startup, and to see when resizing happens. _Don't_ manually resize, so we can see if it's happening automatically or not.

(On my Linux system at least, the initial window size seems to be 800x600 sometimes, but then there are immediately two resize events that go through)

**Comment by Simber1** [comment]
That build fixes the issue for me, I have normal taskbar placement (bottom) and 3x 1920x1080 monitors.

**Comment by jaredkrinke** [comment]
Fixed for me as well, thanks!

**Comment by dabreegster** [comment]
Ah wow, didn't think it would be so easy. Thanks for verifying!

Root cause: "Initial inner window size is PhysicalSize { width: 1920, height: 1017 }, monitor is PhysicalSize { width: 1920, height: 1080 }, scale factor is 1"

On Linux, initial window size is kind of bogus, but a resize event immediately gets fired, probably because the window requests maximized state. On Windows, the window immediately has the correct size; no resize events. But the initial window size I was originally using was the entire monitor, not the inner part of the window.
